chain/jsonrpc/src/lib.rs

```diff
@@ -1355,29 +1351,34 @@ impl JsonRpcHandler {
     }
 }
 
-fn rpc_handler(
+async fn rpc_handler(
     message: web::Json<Message>,
     handler: web::Data<JsonRpcHandler>,
-) -> impl Future<Output = Result<HttpResponse, HttpError>> {
-    let response = async move {
-        let message = handler.process(message.0).await?;
-        Ok(HttpResponse::Ok().json(&message))
+) -> HttpResponse {
+    let message = handler.process(message.0).await;
+    let mut response = if let Message::Response(response) = &message {
+        match &response.result {
+            Ok(_) => HttpResponse::Ok(),
+            Err(err) => match err.error_struct {
+                Some(RpcErrorKind::InternalError(_)) | Some(RpcErrorKind::HandlerError(_)) => {
```

Comment:
@frolvanya @khorolets I have just realized that it turns HandlerError into 500 errors, which is not the right thing to do. Handler errors should be HTTP 200 as execution was totally fine and it is just an application level result, it is not an RPC error

Reply:
Do you want me to open a new PR for this?

Reply:
@frolvanya yes, please. Let's align on the types of errors matching the HTTP status codes: https://github.com/near/nearcore/issues/11792#issuecomment-2241189863

Reply:
@frol can you please check if the new PR is your desired fix or not?